flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/StreamingRuntimeContext.java

```diff
@@ -256,6 +256,20 @@ public <T> org.apache.flink.api.common.state.v2.ValueState<T> getValueState(
         return keyedStateStoreV2.getValueState(stateProperties);
     }
 
+    public <T> org.apache.flink.api.common.state.v2.ListState<T> getListState(
```

Comment:
Should `getValueState/getListState/getMapState` be added to `org.apache.flink.api.common.functions.RuntimeContext`?

Reply:
Whether to expose State v2 to `RuntimeContext` needs more discussion.

It is a mistake that the `StateDescriptor` in runtime module. It should be under `flink-core-api`, I'll move this.

Reply:
Ah... I almost forgot, currently we provide `StateDeclaration` for users to leverage state v2. The `StateDescriptor` need to be moved only if we provide this in DS V1.

Reply:
Since `RuntimeContext` is a public API, We should expose it carefully.
Considering `StateManager` will be exposed in FLIP-410.
I think we could consider it after that. (Maybe exposing to `StateManager` is enough)